source/_components/cover.mqtt.markdown

```diff
@@ -159,6 +159,35 @@ unique_id:
   description: An ID that uniquely identifies this cover. If two covers have the same unique ID, Home Assistant will raise an exception.
   required: false
   type: string
+device:
+  description: 'Information about the device this cover is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works through [MQTT discovery](/docs/mqtt/discovery/) and when [`unique_id`](#unique_id) is set.'
+  required: false
+  type: map
+  keys:
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
+      required: false
+      type: list, string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `"connections": [["mac", "02:5b:26:a8:dc:12"]]`.'
+      required: false
+      type: list, tuple
```

Comment:
`tuple` is not a valid type for the documentation.
Seem like this needs to be a `map` with possible `keys`?

Reply:
No, it's not a map. It's a list of 2-tuples, for example `"[["mac", "02:5b:26:a8:dc:12"], ["mac", "02:5b:26:a8:dc:13"]]` which in context would look like:

```json
{
  "state_topic": "...",
  # ...
  "device": {
    "connections": [["mac", "02:5b:26:a8:dc:12"]]
  }
}
```

So in JSON (this configuration option only works through MQTT discovery, so only JSON) it would *actually* be a list of list, but that would look even weirder in the documentation. And a term to describe a list with fixed size is usually `tuple` - hence a list of tuples.

I can just remove `tuple` from above and keep it as `type: list`. In the end I mean the `type:` field accepts anything as it doesn't care about what `list` or `map` or `string` really means.

Reply:
Aah. I would keep it a `type: list` in that case. Simply because we don't have tuples defined at this point in time... and well... it, in fact, is a list.